### PR TITLE
DASH-2405 [Frontend][Contacts]Create Contacts page

### PR DESCRIPTION
**Pull request by dexterhovis**
https://webcomputing.jira.com/browse/DASH-2405

### CODE REVIEW

File: src/appsco-contacts-page.html [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+<link rel="import" href="../bower_components/neon-animation/neon-animated-pages.html">
+<link rel="import" href="../bower_components/neon-animation/animations/fade-in-animation.html">
+<link rel="import" href="../bower_components/neon-animation/animations/fade-out-animation.html">
+<link rel="import" href="../bower_components/iron-flex-layout/iron-flex-layout.html">
+<link rel="import" href="../bower_components/iron-media-query/iron-media-query.html">
+<link rel="import" href="../bower_components/iron-image/iron-image.html">
+
+<link rel="import" href="components/page/appsco-content.html">
+
+<dom-module id="appsco-directory-page">
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: should be called same as the element `appsco-contacts-page`

File: src/appsco-contacts-page.html [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+                margin-top: 10px;
+            }
+            :host div[resource] > *, :host div[info] > * {
+                padding: 10px;
+                box-sizing: border-box;
+            }
+            :host div[info] {
+                height: 100%;
+            }
+            :host div[info] .account-info {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                border-bottom: 1px solid var(--divider-color);
+                height: 50px;
+                @apply(--layout-flex-none);
+                color: var(--primary-text-color);
+            }
+            appsco-account-security {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not used

File: src/appsco-contacts-page.html [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+                --security-score: {
+                    width: 32px;
+                    height: 32px;
+                    font-size: 14px;
+                };
+            }
+            :host div[info] .details-log {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+                height: calc(100% - 50px - 40px);
+                margin-top: 10px;
+            }
+            :host .details-log paper-tabs {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: note needed

File: src/appsco-contacts-page.html [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+                height: 38px;
+                color: var(--primary-text-color);
+            }
+            :host div[info] .info-details-section {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: note needed

File: src/appsco-contacts-page.html [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+                margin-top: 10px;
+                height: 100%;
+            }
+            .tab-content {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: note needed

File: src/appsco-contacts-page.html [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+                margin-right: -10px;
+                padding-right: 10px;
+                overflow: auto;
+            }
+            :host div[info] .info-actions {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+                padding: 0 10px 10px 10px;
+            }
+            .directory-container {
+                margin-top: 10px;
+            }
+            :host div[resource] > *, :host div[info] > * {
+                padding: 10px;
+                box-sizing: border-box;
+            }
+            :host div[info] {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: note needed

File: src/appsco-contacts-page.html [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+<link rel="import" href="components/page/appsco-content.html">
+
+<dom-module id="appsco-directory-page">
+    <template>
+        <style include="shared-styles webkit-scrollbar-style">
+            :host {
+                display: block;
+
+                --resource-width: 318px;
+                --info-width: 318px;
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+
+                --appsco-list-item-message: {
+                    font-size: 14px;
+                    @apply(--paper-font-common-nowrap);
+                };
+
+                --account-log-appsco-list-item: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+                    padding-top: 14px;
+                    padding-bottom: 8px;
+                };
+
+                --account-details-label: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: note needed

File: src/appsco-contacts-page.html [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+                    font-size: 12px;
+                };
+                --account-details-value: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+                    font-size: 14px;
+                };
+
+                --appsco-account-log-container: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: note needed

File: src/appsco-contacts-page.html [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+                height: 100%;
+                overflow: auto;
+            }
+            :host div[content] > * {
+                padding: 0 10px 10px 10px;
+            }
+            .directory-container {
+                margin-top: 10px;
+            }
+            :host div[resource] > *, :host div[info] > * {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: `:host div[info] > *` is not needed

File: src/appsco-contacts-page.html [hunk 16]
```diff
@@ -0,0 +1,271 @@
...
+                overflow: auto;
+            }
+            :host div[info] .info-actions {
+                padding: 0;
+                position: absolute;
+                bottom: 0;
+                left: 0;
+                right: 0;
+            }
+            :host .edit-button {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+            }
+            :host([tablet-screen]) {
+                --resource-width: 250px;
+                --info-width: 250px;
+            }
+            :host([mobile-screen]) {
+                --resource-width: 100%;
+                --info-width: 100%;
+            }
+            :host .account-icon {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: should be contact-icon

File: src/appsco-contacts-page.html [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+                width: 32px;
+                height: 32px;
+                border-radius: 50%;
+                margin-right: 10px;
+            }
+            :host .account-name {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: should be contact name

File: src/appsco-contacts-page.html [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+                @apply(--paper-font-subhead);
+                font-size: 18px;
+            }
+            :host([screen992]) {
+                --account-basic-info: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: contact instead of account

File: src/appsco-contacts-page.html [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+                    width: 140px;
+                };
+                --account-basic-info-values: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: contact instead of account

File: src/appsco-contacts-page.html [hunk 21]
```diff
@@ -0,0 +1,271 @@
...
+                    width: 140px;
+                };
+
+                --account-auth-type-info: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: contact instead of account

File: src/appsco-contacts-page.html [hunk 22]
```diff
@@ -0,0 +1,271 @@
...
+                    width: 140px;
+                };
+                --account-auth-type-info-values: {
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: contact instead of account

File: src/appsco-contacts-page.html [hunk 23]
```diff
@@ -0,0 +1,271 @@
...
+
+            <div content>
+
+                <div class="directory-container">
+
+                </div>
+
+            </div>
+
+            <div class="flex-vertical" info>
```
Comment (i3or1s) [inline src/appsco-contacts-page.html]: not needed

File: src/appsco-contacts-page.html [hunk 24]
```diff
@@ -0,0 +1,172 @@
...
+
+<dom-module id="appsco-contacts-page">
+    <template>
+        <style include="shared-styles webkit-scrollbar-style">
+            :host {
+                display: block;
+
+                --resource-width: 318px;
+
+                --appsco-list-item-message: {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this style mixin.

File: src/appsco-contacts-page.html [hunk 25]
```diff
@@ -0,0 +1,172 @@
...
+            }
+            .contacts-container {
+                margin-top: 10px;
+            }
+            :host div[resource] > * {
+                padding: 10px;
+                box-sizing: border-box;
+            }
+
+            :host .flex-horizontal {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this style. It is unused.

File: src/appsco-contacts-page.html [hunk 26]
```diff
@@ -0,0 +1,172 @@
...
+                @apply(--layout-horizontal);
+                @apply(--layout-center);
+            }
+            :host .flex-vertical {
+                @apply(--layout-vertical);
+            }
+            :host .flex {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this style. It is unused.

File: src/appsco-contacts-page.html [hunk 27]
```diff
@@ -0,0 +1,172 @@
...
+                @apply(--layout-flex);
+            }
+            :host([tablet-screen]) {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this style.

File: src/appsco-contacts-page.html [hunk 28]
```diff
@@ -0,0 +1,172 @@
...
+                --resource-width: 250px;
+                --info-width: 250px;
+            }
+            :host([mobile-screen]) {
+                --resource-width: 100%;
+                --info-width: 100%;
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: --info-width not needed.

File: src/appsco-contacts-page.html [hunk 29]
```diff
@@ -0,0 +1,172 @@
...
+            }
+            :host([screen992]) {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this block of style.

File: src/appsco-contacts-page.html [hunk 30]
```diff
@@ -0,0 +1,172 @@
...
+            }
+        </style>
+
+        <iron-media-query query="(max-width: 992px)" query-matches="{{ screen992 }}"></iron-media-query>
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this component.

File: src/appsco-contacts-page.html [hunk 31]
```diff
@@ -0,0 +1,172 @@
...
+        <iron-media-query query="(max-width: 800px)" query-matches="{{ tabletScreen }}"></iron-media-query>
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this component.

File: src/appsco-contacts-page.html [hunk 32]
```diff
@@ -0,0 +1,172 @@
...
+
+    </template>
+
+    <script>
+        Polymer({
+            is: 'appsco-contacts-page',
+
+            properties: {
+
+                authorizationToken: {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this property.

File: src/appsco-contacts-page.html [hunk 33]
```diff
@@ -0,0 +1,172 @@
...
+                    type: String,
+                    value: ''
+                },
+
+                /**
+                 * Computed headers.
+                 * It contains authorization token.
+                 */
+                computedHeaders: {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this property.

File: src/appsco-contacts-page.html [hunk 34]
```diff
@@ -0,0 +1,172 @@
...
+                    }
+                },
+
+                mobileScreen: {
+                    type: Boolean,
+                    value: false,
+                    reflectToAttribute: true
+                },
+
+                tabletScreen: {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this property.

File: src/appsco-contacts-page.html [hunk 35]
```diff
@@ -0,0 +1,172 @@
...
+                    type: Boolean,
+                    value: false,
+                    reflectToAttribute: true
+                },
+
+                screen992: {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Remove this property.

File: src/appsco-contacts-page.html [hunk 36]
```diff
@@ -0,0 +1,172 @@
...
+                    value: false
+                }
+            },
+
+            behaviors: [
+                Polymer.NeonAnimatableBehavior
+            ],
+
+            observers: [
+                '_updateScreen(mobileScreen, tabletScreen, screen992)'
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Change to _updateScreen(mobileScreen)

File: src/appsco-contacts-page.html [hunk 37]
```diff
@@ -0,0 +1,172 @@
...
+            ],
+
+            attached: function() {
+                this.pageLoaded = false;
+                if (this.mobileScreen || this.tabletScreen || this.screen992) {
```
Comment (NikolaVu) [inline src/appsco-contacts-page.html]: Change to  if (this.mobileScreen).

**Pull request merged**